### PR TITLE
HTTP/2 README

### PR DESCRIPTION
**Pull request by Scottmitch**
Motivation:
HTTP/2 may require some additional setup and limitations due to the required ciphers by the RFC. We should detail this info in a README to make it easier for users.

Modifications:
- Add a README to codec-http2

Result:
Better docs.
Fixes https://github.com/netty/netty/issues/4522

### CODE REVIEW

**Comment by Scottmitch** [comment]
@nmittler @jchambers - PTAL

File: codec-http2/README.md [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+## TLS with OpenSSL
+
+This is currently the recommended approach for doing TLS with Netty.
+
+### Benefits of using OpenSSL
+
+1. **Speed**: In local testing, we've seen performance improvements of 3x over the JDK. GCM, which is used by the only cipher suite required by the [HTTP/2 RFC](https://tools.ietf.org/html/rfc7540#section-9.2.2), is 10-500x faster.
+2. **Ciphers**: OpenSSL has its own ciphers and is not dependent on the limitations of the JDK. This allows supporting GCM on Java 7.
+3. **ALPN to NPN Fallback**: OpenSSL can support ALPN and NPN simultaneously. The JDK implementation by Netty only supports either ALPN or NPN at any given time and [NPN is only supported in JDK 7](https://wiki.eclipse.org/Jetty/Feature/NPN).
+4. **Version Independence**: does not require using a different library version depending on the JDK update. This is a limitation of the JDK ALPN and NPN implementation used by Netty.
```
Comment (nmittler) [inline codec-http2/README.md]: Maybe `Java Version Independence`?

Comment (Scottmitch) [inline codec-http2/README.md]: done

**Comment by nmittler** [comment]
@Scottmitch LGTM ... I wonder if this should be at the top-level and not under http/2.  It seems that much of this is related to TLS in Netty in general and only parts are specific to HTTP/2.  WDYT?

**Comment by Scottmitch** [comment]
@nmittler - Yes some of this is general TLS / tcnative specific stuff. I think we can remove the tcnative setup stuff and instead link to the wiki which already has this info. After this the focus is more on HTTP/2 and ALPN which may make sense to keep in codec-http2. WDYT after the latest commit?

**Comment by trustin** [comment]
How about writing a wiki page about the requirements of Netty and add a section about HTTP/2, and add a link to that page in our web site and our top README file? A README file in a module directory isn't easy to discover. Also, hosting such information in the wiki could be more contributor-friendly.

**Comment by nmittler** [comment]
@trustin +1

**Comment by Scottmitch** [comment]
Lets move discussion to https://github.com/netty/netty/wiki/Requirements-for-4.x

Feel free to edit / change the wiki now!

**Comment by trustin** [comment]
Great job, @Scottmitch. Let me update the web site.

**Pull request closed without merging**